src/hpa.c

```diff
@@ -383,11 +441,28 @@ hpa_alloc_psset(tsdn_t *tsdn, hpa_shard_t *shard, size_t size) {
 	if (err) {
 		hpdata_unreserve(ps, edata_addr_get(edata),
 		    edata_size_get(edata));
+
 		edata_cache_small_put(tsdn, &shard->ecs, edata);
 
 		shard->nevictions++;
 		malloc_mutex_unlock(tsdn, &shard->mtx);
 		malloc_mutex_unlock(tsdn, &shard->grow_mtx);
+
+		/* We'll do a fake purge; the pages weren't really touched. */
+		hpdata_purge_state_t purge_state;
+		void *purge_addr;
+		size_t purge_size;
+		hpdata_purge_begin(ps, &purge_state);
+		bool found_extent = hpdata_purge_next(ps, &purge_state,
+		    &purge_addr, &purge_size);
+		assert(found_extent);
+		assert(purge_addr == addr);
+		assert(purge_size == size);
+		found_extent = hpdata_purge_next(ps, &purge_state,
+		    &purge_addr, &purge_size);
+		assert(!found_extent);
+		hpdata_purge_end(ps, &purge_state);
```

Comment:
Mind leaving it in for now? A subsequent PR will merge this pathway with another one, so it would make the rebase hard (and we do clean it up eventually).

Reply:
Sounds good. Let's not mess with it then.